### PR TITLE
Dev/database class

### PR DESCRIPTION
**Pull request by TheFactoryMustGrowLarger**
* translated database code to class
* changes to the way config is handled
* new key config key "should_initialize_database" to control if the database should be cleared.

### CODE REVIEW

**Review by Nyxion — APPROVED** [review]

File: site/src/App.js [hunk 1]
```diff
@@ -40,7 +38,7 @@ const NotFound = () => {
         <div className="box" style={{padding: "5px"}}>
             404 PAGE NOT FOUND
             <button className="bb-button" style={{marginTop: "10px"}}>
-                    <Link to="/">Go home, youre durnk.</Link>
+                    <Link to="/">Go home, you're durnk.</Link>
```
Comment (TheFactoryMustGrowLarger) [inline site/src/App.js:41]: is the durnk misspelling intentional?

Comment (Nyxion) [reply site/src/App.js:41]: yes

**Comment by TheFactoryMustGrowLarger** [comment]
@Nyxion think you broke something, when trying to answer quiz question:
```
    raise InvalidQuestionIDException('Bug, could not find question corresponding to %s' % question_uuid)
main.InvalidQuestionIDException: Bug, could not find question corresponding to 0
```

**Comment by TheFactoryMustGrowLarger** [comment]
Looks good now

**Pull request merged**